tests/components/config/test_config_entries.py

```diff
@@ -0,0 +1,295 @@
+    hass.loop.run_until_complete(async_setup_component(hass, 'http', {}))
+    hass.loop.run_until_complete(config_entries.async_setup(hass))
+    yield hass.loop.run_until_complete(test_client(hass.http.app))
+
+
+@asyncio.coroutine
+def test_get_entries(hass, client):
+    """Test get entries."""
+    MockConfigEntry(
+       domain='comp',
+       title='Test 1',
+       source='bla'
+    ).add_to_hass(hass)
+    MockConfigEntry(
+       domain='comp2',
+       title='Test 2',
+       source='bla2',
+       state=core_ce.ENTRY_STATE_LOADED,
+    ).add_to_hass(hass)
+    resp = yield from client.get('/api/config/config_entries/entry')
+    assert resp.status == 200
+    data = yield from resp.json()
+    for entry in data:
+        entry.pop('entry_id')
+    assert data == [
+        {
+            'domain': 'comp',
+            'title': 'Test 1',
+            'source': 'bla',
+            'state': 'not_loaded'
+        },
+        {
+            'domain': 'comp2',
+            'title': 'Test 2',
+            'source': 'bla2',
+            'state': 'loaded',
+        },
+    ]
+
+
+@asyncio.coroutine
+def test_remove_entry(hass, client):
+    """Test removing an entry via the API."""
+    entry = MockConfigEntry(domain='demo')
+    entry.add_to_hass(hass)
+    resp = yield from client.delete(
+        '/api/config/config_entries/entry/{}'.format(entry.entry_id))
+    assert resp.status == 200
+    data = yield from resp.json()
+    assert data == {
+        'require_restart': True
+    }
+
+
+@asyncio.coroutine
+def test_available_flows(hass, client):
+    """Test querying the available flows."""
+    with patch.object(core_ce, 'FLOWS', ['hello', 'world']):
+        resp = yield from client.get(
+            '/api/config/config_entries/flow_handlers')
+        assert resp.status == 200
+        data = yield from resp.json()
+        assert data == ['hello', 'world']
+
+
+############################
+#  FLOW MANAGER API TESTS  #
+############################
+
+
+@asyncio.coroutine
+def test_initialize_flow(hass, client):
+    """Test we can initialize a flow."""
+    class TestFlow(ConfigFlowHandler):
+        @asyncio.coroutine
+        def async_step_init(self, user_input=None):
+            schema = OrderedDict()
+            schema[vol.Required('username')] = str
+            schema[vol.Required('password')] = str
+
+            return self.async_show_form(
+                title='test-title',
+                step_id='init',
+                description='test-description',
+                data_schema=schema,
+                errors={
+                    'username': 'Should be unique.'
+                }
+            )
+
+    with patch.dict(HANDLERS, {'test': TestFlow}):
+        resp = yield from client.post('/api/config/config_entries/flow',
+                                      json={'domain': 'test'})
+
+    assert resp.status == 200
+    data = yield from resp.json()
+
+    assert data['title'] == 'test-title'
```

Comment:
Does it make sense to do an assertion on the whole data object to check that there aren't extra values?

Reply:
This was my different approach to the "We should not compare against generated values". I will go for the `pop` approach again here.